src/accelerate/utils/dataclasses.py

```diff
@@ -415,6 +415,13 @@ class ProjectConfiguration:
         metadata={"help": "The current save iteration."},
     )
 
+    use_node_global_storage: bool = field(
+        default=False,
```

Comment:
One more point, the behavior of deepspeed is that it expects that a multi-node environment uses a shared storage (https://www.deepspeed.ai/docs/config-json/#checkpoint-options & https://huggingface.co/docs/transformers/main_classes/deepspeed#use-of-nonshared-filesystem). We can take this behavior as default.

Reply:
Agreed, we'll also change this to be `save_on_each_node` to align with `TrainingArguments`: https://github.com/huggingface/transformers/blob/v4.33.0/src/transformers/training_args.py#L805-L813